draft-ietf-deleg.md

```diff
@@ -264,61 +264,55 @@ Rest of the Step 2's description is not affected by this document.
 
-In summary, in this specific case, DELEG RR does not create zone cut and is not returned in referral responses, and is not considered authoritative on the parent side of a zone cut. DNSSEC signing is unaffected by this instruction, i.e. no special handling for NSEC type bitmap is necessary.
-
-Two corner cases arise if a zone cut contains DELEG RR and server is answering a DE=0 query:
-
-#### QTYPE=DELEG
-
-DE=0 query indicates that for this particular client DELEG RR type is 'not authoritative' on the parent side.
-
-Let zone file for root zone contain following delegation with NS and DELEG records:
-
-    example. NS    ...
-    example. DELEG ...
+The server MUST copy the value of the DE bit from the query into the response.
+(TODO: not really necessary protocol-wise, but might be nice for monitoring the deployment?)
 
-In this specific case, explicit QTYPE=DELEG query will result in referral response generated by following DELEG unaware rules:
+### DELEG-unaware Clients
 
-    ;; Authority
-    example. NS    ...
+DELEG-unaware clients do not use DELEG records for delegation.
+When a DELEG-aware authoritative server responds to a DELEG-unaware client, any DELEG RR in the response does not create zone cut, is not returned in referral responses, and is not considered authoritative on the parent side of a zone cut.
+Because of this, DELEG-aware authoritative servers MUST answer as if they are DELEG-unaware except for two narrow cases described here.
 
-Other RRs might be included as specified by pre-DELEG protocols.
+#### DELEG-unaware Clients Requesting QTYPE=DELEG
 
-#### DELEG without NS
+In DELEG-unaware clients, records with the DELEG RRtype are not authoritative on the parent side.
+Thus, queries with DE=0 and QTYPE=DELEG MUST result in a legacy referral response.
 
-DELEG unaware client might ask for a name which belongs to a zone delegated only with DELEG RRs but without any NS RRs. Such zone is, by definition, not resolvable for DELEG unaware client. Following pre-DELEG specification, in this case DELEG RR itself cannot create a zone cut and this can result in surprising response where RCODE=NXDOMAIN is returned, with correct proof of non-existance, for subdomains of such zone which does actually exist for DELEG aware clients.
+#### DELEG-unaware Clients with DELEG RRs Present but No NS RRs
 
-In this specific case, authoritative server is RECOMMENDED to supplement DELEG unaware response with Extended DNS Error "New Delegation Only".
+DELEG-unaware clients might ask for a name which belongs to a zone delegated only with DELEG RRs (that is, without any NS RRs).
+Such zone is, by definition, not resolvable for DELEG-unaware clients.
+In this case the DELEG RR itself cannot create a zone cut, and the DELEG-aware authoritative server MUST return a response with an RCODE of NXDOMAIN (along with the DNSSEC proof of non-existance if the query had DO=1).
```

Comment:
This line has changed meaning. The original said that NXDOMAIN will happen for "for subdomains of such zone which does actually exist for DELEG aware clients.". The new text says NXDOMAIN always, which would be incorrect (as in, cannot be DNSSEC-validated) if client asked e.g. for "example. TXT".

Reply:
Second comment: I'm not happy with referencing DO bit at all. I think we better leave the definition what needs to be proven when and to whom to DNSSEC spec.